addons/purchase_stock/models/account_invoice.py

```diff
@@ -82,6 +83,8 @@ def _stock_account_prepare_anglo_saxon_in_lines_vals(self):
                                 move.company_id, valuation_date, round=False,
                             )
                             valuation_total_qty += layers_qty
+                        if valuation_total_qty == 0:
+                            raise UserError(_('Odoo is not able to generate the anglo saxon entries. The total valuation of %s is zero.') % _(line.product_id.name))
```

Comment:
it is better to use `float_zero`

@Abridbus, @simongoffin `_(line.product_id.name))` is horrible and it is better to use `display_name`